scripts/generate_tls_handshake_tests.py

```diff
@@ -13,6 +13,137 @@
 
 from mbedtls_framework import tls_test_case
 from mbedtls_framework import typing_util
+
+from mbedtls_framework.tls_test_case import Side, Version
+
+
+# Assume that a TLS 1.2 ClientHello used in these tests will be at most
+# this many bytes long.
+TLS12_CLIENT_HELLO_ASSUMED_MAX_LENGTH = 255
+
+# Minimum handshake fragment length that Mbed TLS supports.
+TLS_HANDSHAKE_FRAGMENT_MIN_LENGTH = 4
+
+def write_tls_handshake_defragmentation_test(
+        out: typing_util.Writable,
+        side: Side,
+        length: Optional[int],
+        version: Optional[Version] = None
+) -> None:
+    """Generate one TLS handshake defragmentation test.
+
+    :param out: file to write to.
+    :param side: which side is Mbed TLS.
+    :param length: fragment length, or None to not fragment.
+    :param version: protocol version, if forced.
+    """
+    #pylint: disable=chained-comparison,too-many-branches,too-many-statements
+
+    our_args = ''
+    their_args = ''
+
+    if length is None:
+        description = 'no fragmentation, for reference'
+    else:
+        description = 'len=' + str(length)
+    if version is not None:
+        description += ', TLS 1.' + str(version.value)
```

Comment:
Not a big fan of accessing `version.value` here and below. Can we have helper methods instead? For example:
- `__str__` -> `TLS 1.x`
- `openssl_opt` -> `-tls1_x`
- etc.

Reply:
Filed https://github.com/Mbed-TLS/mbedtls-framework/issues/144 for maintainability improvements.